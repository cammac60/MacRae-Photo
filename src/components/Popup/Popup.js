import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './Popup.scss';

export const Popup = () => {

  return (
    <div className="Popup">
    </div>
  )

};

export default Popup;
