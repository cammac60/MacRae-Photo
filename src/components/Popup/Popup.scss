@import '../../variables.scss';

.Popup {
  background-color: rgba(0, 0, 0, 0.8);
  height: 100%;
  position: fixed;
  width: 100%;
  z-index: 100;
}

.popup-inner {
  background-color: #000000;
  border: 1px solid $main-accent-color;
  height: 50%;
  left: 20%;
  position: absolute;
  top: 25%;
  width: 60%;
}
