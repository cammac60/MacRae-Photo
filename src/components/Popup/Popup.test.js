import React from 'react';
import { shallow } from 'enzyme';
import Popup from './Popup';

describe('Popup', () => {

  let wrapper, instance;
  const closePopup = jest.fn();
  const mockProps = {
    closePopup
  };

  beforeEach(() => {
    wrapper = shallow(<Popup {...mockProps}/>);
    instance = wrapper.instance();
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should invoke the closePopup method when the close button is clicked', () => {
    wrapper.find('.popup__button').simulate('click');
    expect(closePopup).toHaveBeenCalled();
  });

});
