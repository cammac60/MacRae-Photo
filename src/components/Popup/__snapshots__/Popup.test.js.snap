// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Should match the snapshot 1`] = `ShallowWrapper {}`;
